### PR TITLE
feat: optimize regex on formula step for edge cases

### PR DESCRIPTION
**Pull request by Sanix-Darker**
## WHAT

fix the regex on formula step for weird edge cases such as : 
```
 'cost' - (price_per_l / [ cost]) + (`price_per_l` / 0) / ("price_per_l" + [ volume_ml  ])
```

PS :  a work on AST for our formulas is ongoing and this sanitizer may be removed, but for now we need it !

### CODE REVIEW

File: server/src/weaverbird/backends/pypika_translator/translators/base.py [hunk 1]
```diff
@@ -803,24 +803,40 @@ def formula(
...
-            # we add quotes on columns for postgresql
+        def _sanitize_formula(formula: str) -> str:
+            """
+            This combined regex will :
+                - remove and replace quotes & brackets to the one
+                  corresponding to the translator.
+                - prevent division by zero by adding NULLIF on all division
+                  expression.
+            Exemple of a weird expression : 'cost' + (`price_per_l` / [alcohol_degree]) / (\"price_per_l\" + [volume_ml]) / 0
+            The sanitizer will return : "cost" + ("price_per_l"/ NULLIF("alcohol_degree", 0))/("price_per_l" + "volume_ml")/ NULLIF(0, 0)
```
Comment (lukapeschke) [inline server/src/weaverbird/backends/pypika_translator/translators/base.py:814]:
```suggestion
            These combined regexes will :
                - remove and replace quotes & brackets with the ones
                  corresponding to the translator.
                - prevent division by zero by adding NULLIF on all division
                  expression.
            Exemple of a weird expression : 'cost' + (`price_per_l` / [alcohol_degree]) / (\"price_per_l\" + [volume_ml]) / 0
            The sanitizer will return : "cost" + ("price_per_l"/ NULLIF("alcohol_degree", 0))/("price_per_l" + "volume_ml")/ NULLIF(0, 0)
```

File: server/src/weaverbird/backends/pypika_translator/translators/base.py [hunk 2]
```diff
@@ -803,24 +803,40 @@ def formula(
...
-                r'([a-zA-Z_a-zA-Z]+)', r'{}\1{}'.format(self.QUOTE_CHAR, self.QUOTE_CHAR), formula
+                r'([a-zA-Z_a-zA-Z]+)',
+                r'{}\1{}'.format(self.QUOTE_CHAR, self.QUOTE_CHAR),
+                formula,
             )
-            formula = formula.replace(self.QUOTE_CHAR + self.QUOTE_CHAR, self.QUOTE_CHAR)
-
-            if '/' in formula:
+            # detect and encapsulate / and %
+            if '/' in step.formula:
```
Comment (lukapeschke) [inline server/src/weaverbird/backends/pypika_translator/translators/base.py:825]: why did you change `if '/' in formula:` to `if '/' in step.formula:` ?

**Review by lukapeschke — COMMENTED** [review]
Code LGTM, a doc suggestion and a question

File: server/src/weaverbird/backends/pypika_translator/translators/base.py (hunk 2 above)
Comment (Sanix-Darker) [reply server/src/weaverbird/backends/pypika_translator/translators/base.py:825]: No particular reason, just wanted to do the check on the formula pipeline from the pipeline directly !
Do you think it's better to rollback that ?

Comment (Sanix-Darker) [reply server/src/weaverbird/backends/pypika_translator/translators/base.py]: updated here : https://github.com/ToucanToco/weaverbird/pull/1427/commits/cae8cfe6efcf3ffaf8997be75b67b30236a355a9

File: server/src/weaverbird/backends/pypika_translator/translators/base.py (hunk 1 above)
Comment (Sanix-Darker) [reply server/src/weaverbird/backends/pypika_translator/translators/base.py]: Updated here : https://github.com/ToucanToco/weaverbird/pull/1427/commits/cae8cfe6efcf3ffaf8997be75b67b30236a355a9

File: server/src/weaverbird/backends/pypika_translator/translators/base.py (hunk 2 above)
Comment (lukapeschke) [reply server/src/weaverbird/backends/pypika_translator/translators/base.py]: Since `formula` has been previously modified, yes. Thanks for the updates!

**Review by lukapeschke — APPROVED** [review]
